cmake/PACE_Docs.cmake

```diff
@@ -1,55 +1,120 @@
-# Build documentation
+#[=======================================================================[.rst:
+PACE_Docs
+-----------------
 
-set(DOCS_ROOT_DIR "${Horace_ROOT}/documentation/user_docs")
-set(DOCS_SOURCE_DIR "${DOCS_ROOT_DIR}/docs")
-set(DOCS_WORK_DIR "${DOCS_ROOT_DIR}/build" CACHE FILEPATH "Directory to put in-progress docs")
-set(DOCS_OUTPUT_DIR "${DOCS_WORK_DIR}/html" CACHE FILEPATH "Directory containing built HTML documentation")
-set(MANUAL_WORK_DIR "${DOCS_WORK_DIR}/latex" CACHE FILEPATH "Directory to build LaTeX sources")
-set(MANUAL_OUTPUT_DIR "${DOCS_WORK_DIR}/latex" CACHE FILEPATH "Directory to put compiled LaTeX manual")
+Build Horace user documentation as either HTML pages or a LaTeX manual
+
+Variables required by the module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``${Horace_ROOT}``
+This is provided by the main CMakeLists.txt
+
+Variables defined by the module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Horace_DOCS_WORK_DIR``
+Working directory where HTML docs are to be built
+
+``Horace_DOCS_OUTPUT_DIR``
+Output directory where HTML docs are placed
+
+``Horace_DOCS_PACK_OUTPUT``
+Output filename (including extension) for compressed docs file
+e.g. ``${CMAKE_CURRENT_BINARY_DIR}/docs.zip``
+
+``Horace_MANUAL_WORK_DIR``
+Working directory where LaTeX docs are to be built
+
+``Horace_MANUAL_OUTPUT_DIR``
+Output directory where LaTeX docs are placed
+
+Targets defined by the module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``docs``
+Build a HTML webpage variant of the user documentation in ``Horace_DOCS_OUTPUT_DIR``
+
+``docs-pack``
+Build HTML docs and compress
+
+``manual``
+Build a LaTeX manual variant of the user documentation
+
+#]=======================================================================]
+
+set(Horace_DOCS_ROOT_DIR "${Horace_ROOT}/documentation/user_docs")
+set(Horace_DOCS_SOURCE_DIR "${Horace_DOCS_ROOT_DIR}/docs")
+set(Horace_DOCS_WORK_DIR "${Horace_DOCS_ROOT_DIR}/build" CACHE FILEPATH "Directory to put in-progress docs")
+set(Horace_DOCS_OUTPUT_DIR "${Horace_DOCS_WORK_DIR}/html" CACHE FILEPATH "Directory containing built HTML documentation")
+set(Horace_MANUAL_WORK_DIR "${Horace_DOCS_WORK_DIR}/latex" CACHE FILEPATH "Directory to build LaTeX sources")
+set(Horace_MANUAL_OUTPUT_DIR "${Horace_DOCS_WORK_DIR}/latex" CACHE FILEPATH "Directory to put compiled LaTeX manual")
```

Comment:
In the interest of having fewer options crowding the CMake GUI, do all these need to be cache variables? Would just having a `Horace_DOCS_BUILD_DIR` cache variable be sufficient? The HTML would always be built in `Horace_DOCS_BUILD_DIR/html` and the LaTeX always built in `Horace_DOCS_BUILD_DIR/latex`. Unless I'm missing a use-case for generating the html/latex in separate places (and in separate places to their in-progress docs).

I'm also not sure what "in-progress" docs are, are these the `make` scripts?